### ISSUE TITLE
Passwords do not match in force_change_password page

### ISSUE THREAD

**Issue by exawizards-koki-fujiwara** [aws-amplify/amplify-flutter#1592]
### Description

The default Authenticator page shows 'Password do not match' even though I typed in the exact same passwords.

I am only testing for iOS device, so I'm not sure if it happens for androids.

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [X] Authenticator
- [ ] DataStore
- [ ] Storage

### Steps to Reproduce

1. Create a user in Cognito (Invite via email, generate passwords, email not verified)
2. Using email address(username) and generated password to signin
3. Screen change to 'Change your password to sign in'
4. Type in password and confirm password
5. Press Change Password button

### Screenshots

<img width="385" alt="passwods" src="https://user-images.githubusercontent.com/47234338/168445067-8df121ab-0837-4920-b44e-ba21123811e0.png">

### Platforms

- [X] iOS
- [ ] Android

### Environment

```bash
[✓] Flutter (Channel stable, 3.0.0, on macOS 12.3.1 21E258 darwin-arm, locale en-JP)
[!] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.66.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability
```

### Dependencies

```bash
Dart SDK 2.17.0
Flutter SDK 3.0.0
hrapp 1.0.0+1

dependencies:
- amplify_api 0.4.5 [amplify_api_plugin_interface amplify_core collection flutter meta plugin_platform_interface]
- amplify_auth_cognito 0.4.5 [flutter amplify_auth_plugin_interface amplify_core amplify_auth_cognito_android amplify_auth_cognito_ios collection plugin_platform_interface]
- amplify_authenticator 0.1.0 [amplify_auth_cognito amplify_auth_plugin_interface amplify_core amplify_flutter collection flutter flutter_localizations intl]
- amplify_flutter 0.4.5 [amplify_analytics_plugin_interface amplify_api_plugin_interface amplify_auth_plugin_interface amplify_core amplify_datastore_plugin_interface amplify_storage_plugin_interface collection flutter json_annotation meta plugin_platform_interface]
- cupertino_icons 1.0.4
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]

transitive dependencies:
- amplify_analytics_plugin_interface 0.4.5 [amplify_core flutter meta]
- amplify_api_plugin_interface 0.4.5 [amplify_core collection flutter json_annotation meta]
- amplify_auth_cognito_android 0.4.5 [flutter]
- amplify_auth_cognito_ios 0.4.5 [amplify_core flutter]
- amplify_auth_plugin_interface 0.4.5 [flutter meta amplify_core]
- amplify_core 0.4.5 [flutter plugin_platform_interface collection date_time_format meta uuid]
- amplify_datastore_plugin_interface 0.4.5 [flutter meta collection amplify_core]
- amplify_storage_plugin_interface 0.4.5 [flutter meta amplify_core]
- characters 1.2.0
- clock 1.1.0
- collection 1.16.0
- crypto 3.0.2 [typed_data]
- date_time_format 2.0.1
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- intl 0.17.0 [clock path]
- json_annotation 4.5.0 [meta]
- material_color_utilities 0.1.4
- meta 1.7.0
- path 1.8.1
- plugin_platform_interface 2.1.2 [meta]
- sky_engine 0.0.99
- typed_data 1.3.1 [collection]
- uuid 3.0.6 [crypto]
- vector_math 2.1.2
```

### Device

iPhone SE 3rd generation

### OS

iOS 15.4

### CLI Version

8.2.0

**Comment by haverchuck** [comment]
@exawizards-koki-fujiwara Can you verify that there isn't an extraneous space at the end of the first password field?

**Comment by Jordan-Nelson** [comment]
@exawizards-koki-fujiwara - A fix was merged to trim the entries from both fields to remove extra spaces. That is most likely what is causing the issue you are seeing. If you have confirmed there are no additional whitespace characters and you still see this issue, please let us know.

**Comment by dnys1** [comment]
This has been released in v0.2.0 of the Authenticator. Please re-open if you continue to experience this issue.
